raiden/network/transport/matrix.py

```diff
@@ -536,22 +531,55 @@ def _network_name(self) -> str:
     def _private_rooms(self) -> bool:
         return bool(self._config.get('private_rooms'))
 
-    def _login_or_register(self):
+    def _login_or_register(self, prev_user_id=None, prev_access_token=None):
+        base_username = to_normalized_address(self._raiden_service.address)
+        _match_user = re.match(
+            f'^@{re.escape(base_username)}.*:{re.escape(self._server_name)}$',
+            prev_user_id or '',
+        )
+        if _match_user:  # same user as before
+            self.log.debug('Trying previous user login', user_id=prev_user_id)
+            self._client.set_access_token(user_id=prev_user_id, token=prev_access_token)
+
+            # test previous user credentials with quick /sync call
+            try:
+                self._client.api.get_devices()
+            except MatrixRequestError as ex:
+                self.log.debug(
+                    'Couldn\'t use previous login credentials, discarding',
+                    prev_user_id=prev_user_id,
+                    _exception=ex,
```

Comment:
Why the underscore? And wouldn't `exc_info=True` be nicer?

Reply:
`exception` is already used as key in structlog's dict, setting it again causes an error. `exc_info=True` prints the whole stacktrace, while here I'd rather have only a one-liner repr of the exception that happened, as it's just a small hint of what happened, not an error which would require the whole traceback for debugging.